tensorflow_federated/python/core/impl/executors/sizing_executor.py

```diff
@@ -59,7 +59,7 @@ def get_type_information(value, type_spec):
       lengths = tf.map_fn(tf.strings.length, value, dtype=tf.int32)
       num_elements = tf.reduce_sum(lengths)
     else:
-      num_elements = type_spec.shape.num_elements()
+      num_elements = tf.size(value)
```

Comment:
There is an implicit assumption here that everything passed through the sizing executor is a materialized value; this does not *necessarily* hold in the TFF execution stack. For example, one could write:

@tff.tf_computation([])
def return_zero():
  return 0

@tff.federated_computation(...)
def do_something_with_zero(...):
  zero = return_zero()
  etc...

This^ would cause a TFF AST representing a called function to be passed down as a tensor type, potentially going through this SizingExecutor.

One option we might have would be: check if num_elements returns None; if so; try to pull this info off the value (or check the value is a tensor or an array); raise if this is not the case, saying sorry, you shouldnt use the sizing executor in a situation where sizes are not concrete. I beleive that this would be a preferable restriction to make, as I think it would only come into existence during some strange situations, e.g. if you wrote (can you even do this?)

@tff.tf_computation([])
def return_unknown_shape(x):
  return tf.reshape([0], [None])

Reply:
I'm actually using this, `type_spec.shape.num_elements` is broken. `tf.size(value)` works but may have an additional issue that you mention, but I don't think it should block this merge.

In any case, IMO throwing an exception in runtime is bad and surprising in this case. If a value is not yet materialized its size should still be recorded eventually somehow.